### PR TITLE
API docs | Docs screenshots | Refactor rescaling | Rearrange widgets

### PR DESCRIPTION
**Pull request by lorenzocerrone**
This PR includes
- [x] first draft of the Python-api documentation 
- [x] some minor sttructure changes
- [x] #246
- [ ] ~~gui video-docs~~ POSTPONED
- [x] new rescaling widget

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -70,10 +70,17 @@ nav:
     - Installation: chapters/getting_started/installation.md
     - Contributing: chapters/getting_started/contributing.md
     - Troubleshooting: chapters/getting_started/troubleshooting.md
-  - Models and Data:
+  - PlantSeg Functionalities:
+    - XXX
+    - Data Processing: XXX
+    - Prediction: XXX
+    - Models: XXX
+    - Segmentation: XXX
```
Comment (qin-yu) [inline mkdocs.yml:78]: Is this intended?

File: mkdocs.yml [hunk 2]
```diff
@@ -44,7 +44,7 @@ theme:
 extra:
   social:
     - icon: fontawesome/brands/github
-      link: https://github.com/pyapp-kit/magicgui
+      link: https://github.com/kreshuklab/plant-seg
```
Comment (qin-yu) [inline mkdocs.yml:47]: Thanks lol

**Review by qin-yu — COMMENTED** [review]
I skimmed through the changes and marked whatever I felt suspicious. Didn't really check line by line, but looks great!

File: plantseg/segmentation/functional/segmentation.py [hunk 3]
```diff
@@ -145,7 +145,7 @@ def mutex_ws(boundary_pmaps: np.ndarray,
     """
     Wrapper around gasp with mutex_watershed as linkage criteria.
 
-    Args:
+    Args:magicgui
```
[the remark this answers is not in the document]
Comment (lorenzocerrone) [reply plantseg/segmentation/functional/segmentation.py:148]: yes thanks for spotting it!

File: mkdocs.yml (hunk 1 above)
Comment (lorenzocerrone) [reply mkdocs.yml:78]: Those are only place holders, I will finalize them before merging

File: mkdocs.yml (hunk 2 above)
Comment (lorenzocerrone) [reply mkdocs.yml:47]: Thanks for looking into it! But it's still on-going! I will ask for a review when ready :)

**Comment by lorenzocerrone** [comment]
Let's merge it as it is now. This PR It's getting to big and too long. I will finish the docs in a next PR

**Comment by qin-yu** [comment]
Yeah I also think this went over the scope of this PR already. If you don't need me to check anything then just merge it.

**Comment by lorenzocerrone** [comment]
I would like you to check the new rescaling widget

**Comment by qin-yu** [comment]
I read the rescaling parts of changes and launched PlantSeg to see it. It looks great, and I think

1. As long the rescaling functionalities are documented, such as the difference between `to_voxel_size` and `set_shape`, then all good
2. To test these, maybe it's better to also write tests for them? It shouldn't rely on us clicking the widget 7 times to test each functions
3. It is only a temporary solution to suppress warning with `# type: ignore`

I can do 2 and 3 if you don't have time for it.

**Comment by qin-yu** [comment]
I feel it make sense to have "Set voxel size" here, but it's still confusing before users read the docs. It is a very useful thing to have and I need it, but I don't know where to put either. If you thought about it already then let's just leave it there.

**Comment by lorenzocerrone** [comment]
> I feel it make sense to have "Set voxel size" here, but it's still confusing before users read the docs. It is a very useful thing to have and I need it, but I don't know where to put either. If you thought about it already then let's just leave it there.

The alternative would be to put it in the IO tab as a stand alone widget.

**Comment by qin-yu** [comment]
I was thinking if we could merge all changes up to ff3d714ad12d06660dde35bf1457e3e7eff1e130 and move the later commits into another PR and work there, but then realised the later commits are kind of related. Since you are done with the planed documentation update in this PR, let's rename and merge it before it includes everything. 

> Yes it would be great if we you could do 2 and 3. Yeah I will document it all

> The alternative would be to put it in the IO tab as a stand alone widget.

I'll do these in the next one.

**Review by qin-yu — APPROVED** [review]

**Pull request merged**